data/data/azure/bootstrap/main.tf

```diff
@@ -90,15 +102,10 @@ resource "azurerm_network_interface" "bootstrap" {
     subnet_id                     = var.subnet_id
     name                          = local.bootstrap_nic_ip_configuration_name
     private_ip_address_allocation = "Dynamic"
+    public_ip_address_id          = "${azurevm_public_ip.bootstrap_public_ip.name}"
```

Comment:
nit: this should be `azurerm_public_ip.bootstrap_public_ip.name`
1. with terraform 0.12, variables don't need quotes https://www.terraform.io/upgrade-guides/0-12.html#first-class-expressions
2. the resource name is `azurerm_public_ip` not `azurevm_public_ip`

Reply:
Fixed!